src/efivar.c

```diff
@@ -424,6 +433,172 @@ prepare_data(const char *filename, uint8_t **data, size_t *data_size)
+	const char *parttype;
+	const char *esp_parttype = "c12a7328-f81f-11d2-ba4b-00a0c93ec93b"; // ESP UUID
+	FILE *mounts;
+	struct mntent *ent;
+	blkid_probe pr = NULL;
+	int r, size = 0;
+	*esp_count = 0;
+
+	// Initialize the blkid cache
+	if (blkid_get_cache(&cache, NULL) < 0) {
+		fprintf(stderr, "Failed to get blkid cache\n");
+		return;
+	}
+
+	// Probe all block devices
+	blkid_probe_all(cache);
+
+	// Create an iterator to go through all block devices
+	iter = blkid_dev_iterate_begin(cache);
+	if (!iter) {
+		fprintf(stderr, "Failed to create blkid iterator\n");
+		blkid_put_cache(cache);
+		return;
+	}
+
+	// Open /proc/mounts to find mount points
+	mounts = setmntent("/proc/mounts", "r");
+	if (!mounts) {
+		fprintf(stderr, "Failed to open /proc/mounts\n");
+		endmntent(mounts);
+		return;
+	}
+
+	while (blkid_dev_next(iter, &dev) == 0) {
+		dev = blkid_verify(cache, dev);
+		if (!dev) {
+			continue;
+		}
+		devname = blkid_dev_devname(dev);
+
+		pr = blkid_new_probe_from_filename(devname);
+		if (!pr) {
+			fprintf(stderr, "Failed to create blkid probe\n");
+			blkid_free_probe(pr);
+			return;
+		}
+
+		// Generate partition details
+		blkid_probe_enable_partitions(pr, 1);
+		blkid_probe_set_partitions_flags(pr, BLKID_PARTS_ENTRY_DETAILS);
+
+		r = blkid_do_safeprobe(pr);
+		if (r == -2) {
+			debug("File system %s is ambiguous\n", devname);
+			continue;
+		} else if (r != 0) {
+			debug("Failed to probe file system %s\n", devname);
+			continue;
+		}
+
+		// Get the PARTTYPE of the current device
+		r = blkid_probe_lookup_value(pr, "PART_ENTRY_TYPE", &parttype, NULL);
+		if (r != 0) {
+			debug("Failed to probe partition type UUID %s\n", devname);
+			continue;
+		}
+
+		if (parttype && strcmp(parttype, esp_parttype) == 0) {
+			// Iterate over mount points to find ESP matches
+			while ((ent = getmntent(mounts)) != NULL) {
+				if (strcmp(ent->mnt_fsname, devname) == 0) {
+					printf("ESP %s path is mounted at dev %s\n", ent->mnt_dir, devname);
+        			size += strlen(ent->mnt_dir) + 1; // +1 for the null terminator
+					char *temp = realloc(*esp_paths, size);
+					if (temp == NULL) {
+						fprintf(stderr, "Failed to allocate memory for ESP paths\n");
+						free(*esp_paths);
+						*esp_count = 0;
+						return;
+					}
+					*esp_paths = temp;
+					strcpy(*esp_paths + size - strlen(ent->mnt_dir) - 1, ent->mnt_dir);
+					(*esp_count)++;
+				}
+			}
+		}
+
+		// Cleanup
+		blkid_free_probe(pr);
+	}
+
+	// Cleanup
+	endmntent(mounts);
+	blkid_dev_iterate_end(iter);
+	blkid_put_cache(cache);
+}
+
+static void
+get_esp_filename(char **esp_filename)
+{
+	char *dst_name = NULL;
+	efi_guid_t dst_guid = efi_guid_empty;
+	size_t filename_size;
+	uint32_t filename_attr;
+	uint8_t *filename = NULL;
+	int rc;
+
+	parse_name(EFI_RTSTORAGEVOLATILE, &dst_name, &dst_guid);
+	if (!dst_name || efi_guid_is_empty(&dst_guid)) {
+		fprintf(stderr, "efivar: could not parse %s\n", EFI_RTSTORAGEVOLATILE);
+		show_errors();
+		exit(1);
```

Comment:
Won't this break for all other platforms? Don't exit if you can't find that special variable. In fact don't even print an error. 
The only case where we print an error should be 
- The RTStorageVolatile was found
- The file is already in ESP
- We failed to copy VarToFile there